internal/api/v1/user/querybuilders/metadata_filter_builder.go

```diff
@@ -0,0 +1,116 @@
+package querybuilders
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"reflect"
+	"strings"
+)
+
+type Metadata map[string]any
+
+const DefaultMaxDepth = 100
+
+type MetadataFilterBuilder struct {
+	MaxDepth int
+	Metadata Metadata
+}
+
+func (m *MetadataFilterBuilder) Build() (url.Values, error) {
+	var sb strings.Builder
+	for k, v := range m.Metadata {
+		var path strings.Builder
+		pref := fmt.Sprintf("metadata[%s]", k)
+		path.WriteString(pref)
+		if err := m.generateQueryParams(0, &path, v, &sb, pref); err != nil {
+			return nil, err
+		}
+	}
+
+	params := make(url.Values)
+	if sb.Len() == 0 {
+		return params, nil
+	}
+
+	pairs := strings.Split(TrimLastAmpersand(sb.String()), "&")
+	for _, pair := range pairs {
+		if len(pair) == 0 {
+			continue
+		}
+
+		split := strings.Split(pair, "=")
```

Comment:
What if I want to filter by a string which contains the `=` character as VALUE?
For example: `metadata = map[string]any{ "someField": "=" }`

What if I want to filter by a string which contains the `=` as  FIELD?
For example `metadata = map[string]any{"my=custom=field": "someValue"}

Reply:
In this case, these types of keys won't be parsed correctly. Based on our previous discussion, the method's caller should have the flexibility to decide how to structure specific filters.

Here's my proposed solution, which I've tested locally and verified that it works. 

We can introduce an option that allows the caller to set the filter using a custom value. Hardcoding a specific value could result in the same issue recurring.

What are your thoughts on this approach? :)

Reply:
Hmmm I would expect that it works without additional options - as it does in `js-client`.
For example, I made such experiment. This test passes in `js-client`:

```ts
test('should build query string with page, filter, and metadata', () => {
    const options: BuildPathOptions = {
      filter: {
        blockHeight: 859864,
      },
      metadata: {
        "hey=123&522": 'example',
        "test": 'value=123'
      },
      page: {
        page: 1,
        pageSize: 1,
        sortDirection: 'desc',
      },
    };

    const result = buildQueryPath(options);

    const p = new URLSearchParams(result);
    expect(p.get("metadata[hey=123&522]")).toBe('example');
    expect(p.get("metadata[test]")).toBe('value=123');
    expect(p.get("page")).toBe('1');
  });
```